### PR TITLE
indent snippets correctly and remove superfluous empty lines

### PR DESCRIPTION
**Pull request by darzu**
Fixes: https://github.com/microsoft/pxt-minecraft/issues/1671
Fixes: https://github.com/microsoft/pxt-minecraft/issues/1667

### CODE REVIEW

**Comment by darzu** [comment]
inserting before

**Comment by darzu** [comment]
With @shakao 's changes from (https://github.com/microsoft/pxt/pull/5753/files):

File: webapp/src/monaco.tsx [hunk 1]
```diff
@@ -761,13 +761,17 @@ export class Editor extends toolboxeditor.ToolboxEditor {
                 ev.preventDefault();
                 ev.stopPropagation();
 
+                // if inline snippet, expects dataTransfer in form "inline:1&qName:name" or "inline:1&[snippet]"
+                let inline = insertText.startsWith("inline:1&");
```
Comment (jwunderl) [inline webapp/src/monaco.tsx]: ie11 :( I think we have a pxtc.util.startsWith

File: webapp/src/monaco.tsx [hunk 2]
```diff
@@ -795,19 +799,95 @@ export class Editor extends toolboxeditor.ToolboxEditor {
...
+        if (insertMode === "NewLineAfter") {
+            // place snippet as if the cursor where at the end of the line
+            position.column = model.getLineMaxColumn(cursorPos.lineNumber)
+        } // else no change
+
+        // if not inline, put snippet on a new line
+        const cursorLineContent = model.getLineContent(cursorPos.lineNumber)
+        const cursorLineIsEmpty = !cursorLineContent
+        if (insertMode === "Inline") {
+            // TODO
```
Comment (jwunderl) [inline webapp/src/monaco.tsx]: is there a note you can add for the todo here / is there still something that needs doing with shannons change in this branch?

Comment (darzu) [reply webapp/src/monaco.tsx]: Actually there was nothing todo so I removed it

**Pull request merged**